src/app/_helpers/factories/building.factory.ts

```diff
@@ -0,0 +1,25 @@
+import { Building} from '../../_models/building.model';
+
+export class BuildingFactory {
+
+  constructor() {
+  }
+
+  createBuildingComponent(id: number, type: string): Building {
```

Comment:
@HonzaDudek it's not creating a component just a model, let's just call it `createBuilding`